sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2965,4 +2965,32 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("SPARK-16602 Nvl/Coalesce") {
```

Comment:
Oh, right. I see. It's the same way to input type checking. This test is too heavy for the exact purpose.